src/Analyzers/Analyzers/src/UseParameterAttributeCorrectlyAnalyzer.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Analyzers.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseParameterAttributeCorrectlyAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_rule = new(
+#pragma warning disable RS2008 // Enable analyzer release tracking
+            id: "ASP0002",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+            title: "A property that has 'ParameterAttribute' should be an auto-property",
+            messageFormat: "The property '{0}' should be auto-property",
+            category: "",
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+        public override void Initialize(AnalysisContext context)
```

Comment:
Can we plug this in to https://github.com/dotnet/aspnetcore/blob/main/src/Components/Analyzers/src/ComponentParameterAnalyzer.cs#L15? We do not need to reinvent parameter discovery.